Remotion/Validation/UnitTests/Rules/PropertyValidationRuleTest.cs

```diff
@@ -1,33 +1,250 @@
+
+      var validatorMock = new Mock<IPropertyValidator>();
+
+      PropertyValidatorContext propertyValidatorContext = null;
+      validatorMock
+          .Setup(_ => _.Validate(It.IsAny<PropertyValidatorContext>()))
+          .Callback((PropertyValidatorContext ctx) => { propertyValidatorContext = ctx; })
+          .Returns(new[] { propertyFailure });
+
+
+      object propertyObj = null;
+      object propertyValueObj = null;
+      Func<object, object> propertyFunc = obj =>
+      {
+        propertyObj = obj;
+        propertyValueObj = "test";
+        return propertyValueObj;
+
+      };
+      object conditionObj = null;
+      Func<object, bool> conditionFunc = obj =>
+      {
+        conditionObj = obj;
+        return true;
+      };
+
+      var rule = new PropertyValidationRule<object, int>(propertyInformationMock.Object, propertyFunc, conditionFunc, new[] { validatorMock.Object });
+
+      var result = rule.Validate(context);
+
+      Assert.That(result, Is.EqualTo(new[] { propertyFailure }));
+      Assert.That(context.InstanceToValidate, Is.SameAs(propertyValidatorContext.Instance));
+      Assert.That(context.InstanceToValidate, Is.SameAs(conditionObj));
+      Assert.That(context.InstanceToValidate, Is.SameAs(propertyObj));
+      Assert.That(propertyValueObj, Is.EqualTo("test"));
+    }
+
+    [Test]
+    public void Validate_WithoutCondition_ReturnsValidationResults ()
+    {
+      var context = new ValidationContext(new Person());
+      var propertyInformationMock = new Mock<IPropertyInformation>();
+
+      var propertyFailure = new PropertyValidationFailure(new Person(), propertyInformationMock.Object, "propval", "err", "validmessage");
+
+      var validatorMock = new Mock<IPropertyValidator>();
+
+      PropertyValidatorContext propertyValidatorContext = null;
+      validatorMock
+          .Setup(_ => _.Validate(It.IsAny<PropertyValidatorContext>()))
+          .Callback((PropertyValidatorContext ctx) => { propertyValidatorContext = ctx; })
+          .Returns(new[] { propertyFailure });
+
+
+      object propertyObj = null;
+      object propertyValueObj = null;
+      Func<object, object> propertyFunc = obj =>
+      {
+        propertyObj = obj;
+        propertyValueObj = "test";
+        return propertyValueObj;
+
+      };
+
+      var rule = new PropertyValidationRule<object, int>(propertyInformationMock.Object, propertyFunc, null, new[] { validatorMock.Object });
+
+      var result = rule.Validate(context);
+
+      Assert.That(result, Is.EqualTo(new[] { propertyFailure }));
+      Assert.That(context.InstanceToValidate, Is.SameAs(propertyValidatorContext.Instance));
+      Assert.That(context.InstanceToValidate, Is.SameAs(propertyObj));
+      Assert.That(propertyValueObj, Is.EqualTo("test"));
+    }
+
+    [Test]
+    public void IsActive_WithoutCondition_WithoutInstanceToValidate_ReturnsTrue ()
+    {
+      var context = new ValidationContext(null);
```

Comment:
This might be nitpicking at a completely over the top level, but I think you should probably move context further down in the tests where you don't need it earlier. 

It would make it easier to see that it is only used when calling the method under test.